src/layout/layout.rs

```diff
@@ -812,53 +961,164 @@ impl Layout {
-        //         .collect::<Vec<&f64>>()
-        // );
-        // dbg!(ends);
-
+    fn get_results(
+        changes: &HashMap<Variable, f64>,
+        area: Rect,
+        layout: &Layout,
+        elements: &[Element],
+    ) -> Rects {
         // convert to Rects
-        let results = elements
+        elements
             .iter()
             .map(|element| {
                 let start = changes.get(&element.start).unwrap_or(&0.0).round() as u16;
                 let end = changes.get(&element.end).unwrap_or(&0.0).round() as u16;
-                let size = end - start;
+                let size = end.saturating_sub(start);
                 match layout.direction {
                     Direction::Horizontal => Rect {
                         x: start,
-                        y: inner.y,
+                        y: area.y,
                         width: size,
-                        height: inner.height,
+                        height: area.height,
                     },
                     Direction::Vertical => Rect {
-                        x: inner.x,
+                        x: area.x,
                         y: start,
-                        width: inner.width,
+                        width: area.width,
                         height: size,
                     },
                 }
             })
-            .step_by(if spacers_added { 2 } else { 1 })
-            .collect::<Rc<[Rect]>>();
-        Ok(results)
+            .collect::<Rects>()
+    }
+
+    fn try_split(area: Rect, layout: &Layout) -> Result<(Rects, Rects), AddConstraintError> {
+        let strengths = StrengthSet::default();
+        if !strengths.is_valid() {
+            return Err(AddConstraintError::InternalSolverError(
+                "Constraint strengths will result in unstable layouts",
+            ));
+        }
+
+        // To take advantage of all of cassowary features, we would want to store the `Solver` in
+        // one of the fields of the Layout struct. And we would want to set it up such that we could
+        // add or remove constraints as and when needed.
+        // The advantage of doing it as described above is that it would allow users to
+        // incrementally add and remove constraints efficiently.
+        // Solves will just one constraint different would not need to resolve the entire layout.
+        //
+        // The disadvantage of this approach is that it requires tracking which constraints were
+        // added, and which variables they correspond to.
+        // This will also require introducing and maintaining the API for users to do so.
+        //
+        // Currently we don't support that use case and do not intend to support it in the future,
+        // and instead we require that the user re-solve the layout every time they call `split`.
+        // To minimize the time it takes to solve the same problem over and over again, we
+        // cache the `Layout` struct along with the results.
```

Comment:
I put this in here mainly because if someone wants to refactor cassowary in the future or improve the implementation, they may try to make it such that we create an instance of the solver just once; and I wanted to preemptively hedge against that.